app/components/Instructions/Instructions.js

```diff
@@ -2,57 +2,55 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 export default function Instructions({ state, }) {
-  function buildInstructionArray(appState) {
+  function createInstructionArray(appState) {
     const instructionList = [];
     const {
       pathStartingPoint,
       pathEndingPoint,
       startButtonPressed,
       endButtonPressed,
     } = appState;
-    const setGoBackText = buttonText => (
+    const getGoBackText = buttonText => (
       `Press on the "${buttonText}" button in order to go back to toggle mode`
     );
-    const setReferencePoint = buttonText => (
+    const getReferencePoint = buttonText => (
       `Press on a tile to set it as the "${buttonText}" of the path`
     );
-    const gameInstructions = {
-      toggleTile: 'Press on tiles to toggle tile availability',
-      goToChooseStartMode: 'Press the "START" button to set the starting point',
-      goToChooseEndMode: 'Press the "END" button to set the ending point',
-      chooseReferencePoint: setReferencePoint(startButtonPressed ? 'Start' : 'End'),
-      goBackToToggleTileMode: setGoBackText(startButtonPressed ? 'START' : 'END'),
-      findAndDrawPath: 'Press the "GO!" button to draw path',
-    };
+    const toggleTile = 'Press on tiles to toggle tile availability';
+    const goToChooseStartMode = 'Press the "START" button to set the starting point';
+    const goToChooseEndMode = 'Press the "END" button to set the ending point';
+    const chooseReferencePoint = getReferencePoint(startButtonPressed ? 'Start' : 'End');
+    const goBackToToggleTileMode = getGoBackText(startButtonPressed ? 'START' : 'END');
+    const findAndDrawPath = 'Press the "GO!" button to draw path';
     if (!startButtonPressed && !endButtonPressed) {
-      instructionList.push(gameInstructions.toggleTile);
+      instructionList.push(toggleTile);
       if (pathStartingPoint.length === 0) {
-        instructionList.push(gameInstructions.goToChooseStartMode);
+        instructionList.push(goToChooseStartMode);
       }
       if (pathEndingPoint.length === 0) {
-        instructionList.push(gameInstructions.goToChooseEndMode);
+        instructionList.push(goToChooseEndMode);
       }
       if (pathStartingPoint.length !== 0 && pathEndingPoint.length !== 0) {
-        instructionList.push(gameInstructions.findAndDrawPath);
+        instructionList.push(findAndDrawPath);
       }
     }
     if (startButtonPressed || endButtonPressed) {
-      instructionList.push(gameInstructions.chooseReferencePoint);
-      instructionList.push(gameInstructions.goBackToToggleTileMode);
+      instructionList.push(chooseReferencePoint);
+      instructionList.push(goBackToToggleTileMode);
```

Comment:
All these `push`es can be DRYed and made readable: `const addInstruction = instructionList.push`. You may need to use bind too though: `const addInstruction = instructionList.push.bind(instructionList)`

Reply:
done